packages/engine-core/src/library/ExitHooks.ts

```diff
@@ -48,8 +49,8 @@ export class ExitHooks {
     return this.idToListenerMap.get(id)
   }
 
-  private installHook(event: string, shouldExit = false) {
-    process.once(event, async (code) => {
+  private installHook(event: 'beforeExit' | 'exit' | NodeJS.Signals, shouldExit = false) {
```

Comment:
I think by that point, there is very little in common between the handler for signals and `exit`/`beforeExit` event and we would make this code much simpler by splitting `installHook` into two methods, one for exit events and one for signals. I needed to deal with `ExitHooks` problem in my other PR and took this approach: https://github.com/prisma/prisma/pull/18969/files#diff-24df258365237f8cecaaabdd47cbad94bf43846b3fbea549316e93f23d338505

Reply:
Thanks for the feedback, I'll be splitting this `installHook` in two. At this point, I think we can also get rid of the `shouldExit` argument, and always apply the `shouldExit` callback on the hooks installed by `SIG*` signals. What do you think?

Reply:
Yep, `shouldExit` is always `true` for signals and always `false` for exit events, so totally makes sense 👍